### PR TITLE
[mpg123] Modernize for Conan 2.0

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **mpg123/1.31.2**

Related to https://github.com/conan-io/conan-center-index/pull/15965

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Pkg-config should be available now

File: recipes/mpg123/all/conanfile.py [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import cmake_layout, CMake, CMakeDeps, CMakeToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conan.tools.files import get, copy, export_conandata_patches, apply_conandata_patches, rmdir, rm
+from conan.tools.microsoft import is_msvc
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.build import cross_building
+from conan.tools.apple import fix_apple_shared_install_name
```
Comment (prince-chrismc) [inline recipes/mpg123/all/conanfile.py:10]:
```suggestion
```

File: recipes/mpg123/all/conanfile.py [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.53.0"
 
 
 class Mpg123Conan(ConanFile):
     name = "mpg123"
     description = "Fast console MPEG Audio Player and decoder library"
-    topics = ("conan", "mpg123", "mpeg", "audio", "player", "decoder")
+    topics = ("mpg123", "mpeg", "audio", "player", "decoder")
```
Comment (prince-chrismc) [inline recipes/mpg123/all/conanfile.py:19]:
```suggestion
    topics = ("mpeg", "audio", "player", "decoder")
```

**Comment by uilianries** [comment]
@prince-chrismc Thank you for reviewing! 😄

File: recipes/mpg123/all/conanfile.py [hunk 3]
```diff
@@ -43,153 +50,162 @@ class Mpg123Conan(ConanFile):
...
+        if self.settings.os != "Windows" and self.options.module == "win32":
+            raise ConanInvalidConfiguration(f"The option -o {self.ref.name}:module should not use 'win32' for non-Windows OS")
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
```
Comment (Croydon) [inline recipes/mpg123/all/conanfile.py:97]: Why are we adding such switches again? In the past, we had the opinion that such switches only introduce trouble

Comment (uilianries) [reply recipes/mpg123/all/conanfile.py:97]: configuration does not affect the package, so users can add or not without affect the resulting package. But on Conan 2.x, tool_requires affect the package ID, so we can reproduce more cases, or at least, know when users are using or not build requirements.

Comment (Croydon) [reply recipes/mpg123/all/conanfile.py:97]: I would still prefer to control the tool_requires globally

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/mpg123/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -1,10 +1,7 @@
...
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES C)
 
 find_package(mpg123 REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} PRIVATE MPG123::libmpg123)
+target_link_libraries(${PROJECT_NAME} PRIVATE MPG123::libmpg123 MPG123::libout123 MPG123::libsyn123)
```
Comment (jcar87) [inline recipes/mpg123/all/test_package/CMakeLists.txt:7]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE MPG123::libmpg123)
```

Here I think `MPG123::libmpg123` alone is correct - they're 3 independent libraries (I can't see any dependencies between them), with 3 distinct header files - if the `test_package.c` file is only calling functionality from `mpg123.h` then this one suffices. If we add more libraries, on newer Linux distros `-as-needed` is passed and they will be eventually ignored, and on macOS the test executable will require the other two libraries at runtime (unnecessarily).

File: recipes/mpg123/all/conanfile.py [hunk 5]
```diff
@@ -43,153 +50,162 @@ class Mpg123Conan(ConanFile):
...
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
             autotools.install()
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+            rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
```
Comment (jcar87) [inline recipes/mpg123/all/conanfile.py:179]:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
        
        fix_apple_shared_install_name(self)
```

when `shared=True` the libraries had absolute path in their install name :D

File: recipes/mpg123/all/conanfile.py [hunk 6]
```diff
@@ -1,14 +1,21 @@
-from conans import AutoToolsBuildEnvironment, CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import cmake_layout, CMake, CMakeDeps, CMakeToolchain
```
Comment (jcar87) [inline recipes/mpg123/all/conanfile.py:3]:
```suggestion
from conan.tools.apple import fix_apple_shared_install_name
from conan.tools.cmake import cmake_layout, CMake, CMakeDeps, CMakeToolchain
```

see comment below

**Review by jcar87 — DISMISSED** [review]
Amazing!!!

**Comment by uilianries** [comment]
Failed download. Restarting CI

**Review by jcar87 — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Pull request merged**